src/reuser/ui/agent-reuser.php

```diff
@@ -114,22 +114,26 @@ public function scheduleAgent($jobId, $uploadId, &$errorMsg, $request)
     }
     $groupId = $request->get('groupId', Auth::getGroupId());
 
-    $getReuseValue = $request->get('reuseMode');
-
+    $getReuseValue = $request->get('reuseMode') ?: array();
     $reuseMode = UploadDao::REUSE_NONE;
-
-    if (! empty($getReuseValue)) {
-      if (count($getReuseValue) < 2) {
-        if (in_array('reuseMain', $getReuseValue)) {
-          $reuseMode = UploadDao::REUSE_MAIN;
-        } else {
-          $reuseMode = UploadDao::REUSE_ENHANCED;
-        }
-      } else {
-        $reuseMode = UploadDao::REUSE_ENH_MAIN;
+    foreach ($getReuseValue as $currentReuseValue) {
+      switch ($currentReuseValue) {
+        case 'reuseMain':
+          $reuseMode |= UploadDao::REUSE_MAIN;
+          break;
+        case 'reuseEnhanced':
+          $reuseMode |= UploadDao::REUSE_ENHANCED;
+          break;
+        case 'reuseConf':
+          $reuseMode |= UploadDao::REUSE_CONF;
+          break;
       }
     }
 
+    if (empty($reuseMode)) {
+      return 0;
```

Comment:
If the user wants to simply reuse the decisions from older package without additional things like main license, the reuser does not get scheduled.